File: api/springBoot-simulator/src/main/java/com/bcsimulator/model/CsvFile.java
package com.bcsimulator.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Table(name = "csv_file")
@Data
public class CsvFile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String path;
    private LocalDateTime createdAt;

    @Column(columnDefinition = "TEXT")
    private String columns; // es: "time,value1,value2,value3"

    @Column(columnDefinition = "TEXT")
    private String configurationJson; // Stores the simulation configuration JSON
}

File: useCases/sesame/client/src/components/types/CsvFile.ts
export interface CsvFile {
    id: string;
    name: string;
    path: string;
    columns: string[];
    createdAt: string;
    configurationJson: string;
}

export interface DatasetRequest {
    fileId: number;
    columns: string[];
}

export interface ChartRequest {
    datasets: DatasetRequest[];
}

export interface ChartResponse {
    labels: string[];
    series: {
        label: string;
        data: number[];
    }[];
}